### PR TITLE
Using a scale transform when writing the video to crop edge artifacts…

### PR DESCRIPTION
**Pull request by brettlyon**
… in iOS 11

### CODE REVIEW

**Review by nas887 — APPROVED** [review]
🤷‍♂️

File: Source/PBJMediaWriter.m [hunk 1]
```diff
@@ -197,7 +197,10 @@ - (BOOL)setupVideoWithSettings:(NSDictionary *)videoSettings withAdditional:(NSD
 
         _assetWriterVideoInput = [AVAssetWriterInput assetWriterInputWithMediaType:AVMediaTypeVideo outputSettings:videoSettings];
         _assetWriterVideoInput.expectsMediaDataInRealTime = YES;
-        _assetWriterVideoInput.transform = CGAffineTransformIdentity;
+        
+        // brightwheel fix - we're scaling up to crop at least 16 px from the edge to remove encoding artifacts that appear in iOS 11.
+        // 1.05 is the value that should ensure at least 16 px cropped on each side on the smallest screen width: 640 px (1 + ((16 * 2)) / 640))
+        _assetWriterVideoInput.transform = CGAffineTransformScale(CGAffineTransformIdentity, 1.05, 1.05);
```
Comment (kamranpirwani) [inline Source/PBJMediaWriter.m:203]: @brettlyon We should limit the scope of this transform only when the actual device bounds are not divisible by 16. We can determine this from the videoSettings dictionary that is passed in.

In short, we don't want to effect the quality for devices that do not currently experience this issue

**Comment by kamranpirwani** [comment]
@brettlyon See comment above ^^^

**Comment by GuillaumeDelente** [comment]
Beaten at the code review game

**Pull request merged**